From 036184123512b1b67243511b14cedd9478a2a49b Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@gmail.com>
Date: Sat, 15 Feb 2025 07:37:01 -0500
Subject: [PATCH] Le Potato general patch set

Change model name
Increase the max sdcard freq
Add power management and vrtc
Add i2c_AO node

Original source: Libre Computer

Signed-off-by: Patrick Yavitz <pyavitz@gmail.com>
---
 arch/arm64/boot/dts/amlogic/meson-gx.dtsi         |  9 +++++++++
 .../dts/amlogic/meson-gxl-s905x-libretech-cc.dts  | 15 ++++++++++++---
 2 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
index 2673f0dbafe7..81ac2bff38fe 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
@@ -223,6 +223,10 @@ sm: secure-monitor {
 		};
 	};
 
+	system-suspend {
+		compatible = "amlogic,meson-gx-pm";
+	};
+
 	efuse: efuse {
 		compatible = "amlogic,meson-gx-efuse", "amlogic,meson-gxbb-efuse";
 		#address-cells = <1>;
@@ -461,6 +465,11 @@ clkc_AO: clock-controller {
 				};
 			};
 
+			vrtc: rtc@a8 {
+				compatible = "amlogic,meson-vrtc";
+				reg = <0x0 0x000a8 0x0 0x4>;
+			};
+
 			cec_AO: cec@100 {
 				compatible = "amlogic,meson-gx-ao-cec";
 				reg = <0x0 0x00100 0x0 0x14>;
diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
index 401064b0428d..9de1ba3d2b03 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
@@ -15,11 +15,13 @@
 / {
 	compatible = "libretech,aml-s905x-cc", "amlogic,s905x",
 		     "amlogic,meson-gxl";
-	model = "Libre Computer AML-S905X-CC";
+	model = "Libre Computer Le Potato";
 
 	aliases {
-		serial0 = &uart_AO;
 		ethernet0 = &ethmac;
+		i2c1 = &i2c_AO;
+		rtc1 = &vrtc;
+		serial0 = &uart_AO;
 	};
 
 	dio2133: analog-amplifier {
@@ -311,7 +313,7 @@ &sd_emmc_b {
 
 	bus-width = <4>;
 	cap-sd-highspeed;
-	max-frequency = <50000000>;
+	max-frequency = <100000000>;
 	disable-wp;
 
 	cd-gpios = <&gpio CARD_6 GPIO_ACTIVE_LOW>;
@@ -345,6 +347,13 @@ &uart_AO {
 	pinctrl-names = "default";
 };
 
+&i2c_AO {
+	/* Pins 3 (SDA), 5 (SCL) */
+	status = "disabled";
+	pinctrl-0 = <&i2c_ao_pins>;
+	pinctrl-names = "default";
+};
+
 &usb {
 	status = "okay";
 	dr_mode = "host";
-- 
2.39.5

